caddytest/caddytest.go

```diff
@@ -78,9 +79,28 @@ func NewTester(t testing.TB) *Tester {
 		},
 		configLoaded: false,
 		t:            t,
+		config:       Default,
 	}
 }
 
+func (tc *Tester) WithDefaultOverrides(config Config) *Tester {
```

Comment:
Can you add a godoc comment on this? And maybe rename `config Config` to `overrides Config` or something. :+1: